crux-core/src/crux/api.clj

```diff
@@ -448,19 +248,170 @@
+    - `:crux.tx/tx-time` (Date):
+        Shorthand for `{:crux.tx/tx {:crux.tx/tx-time <>}}`
+
+  Providing both `:crux.tx/tx` and `:crux.tx/tx-time` is undefined.
+  Arities passing dates directly (`node vt` and `node vt tt`) are deprecated and will be removed in a later release.
+
+  If the node hasn't yet indexed a transaction at or past the given transaction, this throws NodeOutOfSyncException
+
+  This DB opens up shared resources to make multiple requests faster - it must
+  be `.close`d when you've finished using it (for example, in a `with-open`
+  block)"))
+
+(let [db-args '(^java.io.Closeable [node]
+                ^java.io.Closeable [node db-basis]
+                ^java.io.Closeable ^:deprecated [node valid-time]
+                ^java.io.Closeable ^:deprecated [node valid-time tx-time])]
+  (alter-meta! #'db assoc :arglists db-args)
+  (alter-meta! #'open-db assoc :arglists db-args))
+
+(defn q
+  "q[uery] a Crux db.
+  query param is a datalog query in map, vector or string form.
+  This function will return a set of result tuples if you do not specify `:order-by`, `:limit` or `:offset`;
+  otherwise, it will return a vector of result tuples."
+  [db q & args]
+  (q* db q (object-array args)))
+
+(defprotocol TransactionFnContext
+  (indexing-tx [tx-fn-ctx]))
+
+(defn open-q
+  "lazily q[uery] a Crux db.
+  query param is a datalog query in map, vector or string form.
+
+  This function returns a Cursor of result tuples - once you've consumed
+  as much of the sequence as you need to, you'll need to `.close` the sequence.
+  A common way to do this is using `with-open`:
+
+  (with-open [res (crux/open-q db '{:find [...]
+                                    :where [...]})]
+    (doseq [row (iterator-seq res)]
+      ...))
+
+  Once the sequence is closed, attempting to iterate it is undefined."
+  ^crux.api.ICursor [db q & args]
+  (open-q* db q (object-array args)))
+
+(let [arglists '(^crux.api.ICursor
+                 [db eid sort-order]
+                 ^crux.api.ICursor
+                 [db eid sort-order {:keys [with-docs? with-corrections?]
+                                     {start-vt :crux.db/valid-time
+                                      start-tt :crux.tx/tx-time
+                                      start-tx-id :crux.tx/tx-id} :start
+                                     {end-vt :crux.db/valid-time
+                                      end-tt :crux.tx/tx-time
+                                      end-tx-id :crux.tx/tx-id} :end}])]
+  (alter-meta! #'entity-history assoc :arglists arglists)
+  (alter-meta! #'open-entity-history assoc :arglists arglists))
+
+(defrecord JQueryDatasource [^java.io.Closeable datasource]
+  ICruxDatasource
+  (entity [_ eid] (entity datasource eid))
+  (entityTx [_ eid] (entity-tx datasource eid))
+  (query [_ query args] (q* datasource query args))
+  (openQuery [_ query args] (open-q* datasource query args))
+  (entityHistory [_ eid opts] (entity-history datasource eid opts))
```

Comment:
as discussed, don't need the `HistoryOptions` Java object coming all the way through into Clojure now - can adapt it at this point